### PR TITLE
Calculate gas for built in functions

### PR DESCRIPTION
**Pull request by miiu96**
- Calculate the gas cost for built-in functions like ESDT Transfer,ESDT Local Burn,ESDT Local Mint, etc

### CODE REVIEW

File: process/economics/interface.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package economics
+
+import (
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+// BuiltInFunctionsCostHandler is able to calculated the cost of a built in function call
+type BuiltInFunctionsCostHandler interface {
+	ComputeBuiltInCost(tx process.TransactionWithFeeHandler) uint64
+	IsSpecialBuiltIn(tx process.TransactionWithFeeHandler) bool
```
Comment (sasurobert) [inline process/economics/interface.go:10]: add IsInterfaceNIl

File: process/economics/builtInFunctionsCost.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	totalLen := uint64(0)
+	for _, arg := range arguments {
+		totalLen += uint64(len(arg))
+	}
+
+	return totalLen
+}
+
+// IsSpecialBuiltIn will check is the provided transaction is a special build in function call
+func (bc *builtInFunctionsCost) IsSpecialBuiltIn(tx process.TransactionWithFeeHandler) bool {
```
Comment (sasurobert) [inline process/economics/builtInFunctionsCost.go:127]: rename to IsBuiltInFuncCall ?

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: process/economics/interface.go (hunk 1 above)
Comment (miiu96) [reply process/economics/interface.go:10]: done.

File: process/economics/builtInFunctionsCost.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	args.GasSchedule.RegisterNotifyHandler(bs)
+
+	return bs, nil
+}
+
+func (bc *builtInFunctionsCost) initSpecialBuiltInFunctionCostMap() {
```
Comment (AdoAdoAdo) [inline process/economics/builtInFunctionsCost.go:51]: rename?
"special" is too generic, what kind of special?

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

File: process/economics/builtInFunctionsCost.go (hunk 2 above)
Comment (miiu96) [reply process/economics/builtInFunctionsCost.go:127]: done.

File: process/economics/interface.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (miiu96) [reply process/economics/interface.go:10]: removed special from the method name. this function just check if a transaction is built-in function call

File: process/economics/builtInFunctionsCost.go (hunk 3 above)
Comment (miiu96) [reply process/economics/builtInFunctionsCost.go:51]: removed special

File: integrationTests/mock/builtInCostHandlerStub.go [hunk 4]
```diff
@@ -13,7 +13,12 @@ func (b BuiltInCostHandlerStub) ComputeBuiltInCost(_ process.TransactionWithFeeH
 	return 1
 }
 
-// IsSpecialBuiltIn -
-func (b BuiltInCostHandlerStub) IsSpecialBuiltIn(_ process.TransactionWithFeeHandler) bool {
+// IsBuiltInFuncCall -
+func (b BuiltInCostHandlerStub) IsBuiltInFuncCall(_ process.TransactionWithFeeHandler) bool {
```
Comment (AdoAdoAdo) [inline integrationTests/mock/builtInCostHandlerStub.go:17]: can you change to pointer receiver everywhere?

File: epochStart/mock/builtInCostHandlerStub.go [hunk 5]
```diff
@@ -13,7 +13,12 @@ func (b BuiltInCostHandlerStub) ComputeBuiltInCost(_ process.TransactionWithFeeH
 	return 1
 }
 
-// IsSpecialBuiltIn -
-func (b BuiltInCostHandlerStub) IsSpecialBuiltIn(_ process.TransactionWithFeeHandler) bool {
+// IsBuiltInFuncCall -
+func (b BuiltInCostHandlerStub) IsBuiltInFuncCall(_ process.TransactionWithFeeHandler) bool {
```
Comment (AdoAdoAdo) [inline epochStart/mock/builtInCostHandlerStub.go:17]: can you change to pointer receiver everywhere?

File: integrationTests/mock/builtInCostHandlerStub.go (hunk 4 above)
Comment (miiu96) [reply integrationTests/mock/builtInCostHandlerStub.go]: done.

File: epochStart/mock/builtInCostHandlerStub.go (hunk 5 above)
Comment (miiu96) [reply epochStart/mock/builtInCostHandlerStub.go]: done.

**Review by sasurobert — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**